# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.h
# Commit: 622a66cdac2e
# Full Hash: 622a66cdac2e19f54a0d3ef4777380e849a60bf1
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 10: Implement alertfinished behavior r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227695
# ==============================================================================

diff -r 33642b1c547b -r 622a66cdac2e dom/notification/NotificationParent.h
--- a/dom/notification/NotificationParent.h	Thu Nov 07 17:32:07 2024 +0000
+++ b/dom/notification/NotificationParent.h	Thu Nov 07 17:32:07 2024 +0000
@@ -32,7 +32,9 @@
         mIsSecureContext(aIsSecureContext),
         mId(aId),
         mScope(aScope),
-        mOptions(aOptions) {};
+        mOptions(aOptions) {
+    MOZ_ASSERT(!mScope.IsEmpty(), "Only for persistent notifications for now");
+  };
 
   IPCResult RecvShow(ShowResolver&& aResolver);
 
@@ -47,6 +49,7 @@
 
   nsresult Show();
   nsresult FireClickEvent();
+  nsresult FireCloseEvent();
 
   void GetAlertName(nsAString& aRetval) {
     if (mAlertName.IsEmpty()) {
